### DISCUSSION TITLE
No Fail Mod Interaction (Fail At % Accuracy)

### DISCUSSION THREAD

**Discussion by Kevin2Holt** [ppy/osu#28709] [Ideas]
As a super casual player, I much prefer to just play through an entire song every time rather than failing and starting over halfway through. So, I play with No Fail (NF) on pretty much all the time. I've seen another mod called "Accuracy Challenge" (AC) and had the thought of combining the two.

### My Suggestion:
**Add an interaction between NF and AC to allow failing by accuracy, but never by energy/hp.**

I am assuming (I haven't actually tried it) that AC still fails you by energy/hp (the normal fail).

I'm not sure how much explaining I need to do for this idea... Basically, instead of having them be "incompatible", allow AC to "override" (of-sorts) NF. Where NF keeps you from failing if your energy/hp falls too low, but if your accuracy drops below the threshold, AC still fails you.

**Comment by NotGumballer91** [comment]
NoFail entirely prevents failing which Accuracy Challenge does. Perhaps maybe you should use Difficulty Adjust with HP0 instead?

**Reply by Kevin2Holt** [reply]
That's why I suggested the **\*interaction\***, which slightly alters how it works.

Currently, the two mods are "incompatible", so you can't even have them both on at the same time. My suggestion is to allow both (add a "compatibility"), creating an effect like having HP0 _(I didn't think about that before)_, but still allowing you to see what your hp _"would be"_. 

One possible way to make this work is by simply changing the NF mod to only remove failing by energy/hp. AC would then add a new condition for failing. So, with NF only removing the one fail condition (not the one AC adds), you end up with the desired result.

To me, this makes more sense and is more straightforward than needing to understand the hp system (what the numbers mean) and realizing that you need to use HP0 (hidden behind Difficulty Adjust) just to achieve failing by accuracy only.

**Comment by shiumano** [comment]
#28639 Like this?
Well, this was rejected because my program was terrible lol.

**Comment by Sleepteiner** [comment]
There are plenty of maps that I can play decently well, but I can't pass consistently due to a couple of small sections in the map. On maps like those, instead of dealing with the huge waste of time that is the osu pass/fail system, I opt to just use NF like Kevin2Holt does. However, if I use NF on those maps, then I lose the ability to use "standard accuracy mode" Accuracy Challenge to significantly increase efficiency when trying to improve on my previous best scores.

So, I am very in favor of having NF and Accuracy Challenge be compatible.
